eslint / flow integration issues

eg.
```
src/core/index.js
  24:29  error  Spread must be the final element of an element list
```

Why do we need to use `esprima-fb` ? Something to do with flow? I saw on the #384 pull request a change to use the `babel-eslint` parser. 

@wltsmrz  says:
```
babel-eslint seems to be aware of flow https://github.com/babel/babel-eslint/commit/fcd22f553c2037a9eb3c92e8d5d8af4bb7b6bc31
```

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/23525189-eslint-flow-integration-issues?utm_campaign=plugin&utm_content=tracker%2F323756&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F323756&utm_medium=issues&utm_source=github).
</bountysource-plugin>

--- a/bin/ci.sh
+++ b/bin/ci.sh
@@ -11,7 +11,7 @@ typecheck() {
 lint() {
   REPO_URL="https://raw.githubusercontent.com/ripple/javascript-style-guide"
   curl "$REPO_URL/es6/eslintrc" > ./eslintrc
-  echo "parser: esprima-fb" >> ./eslintrc
+  echo "parser: babel-eslint" >> ./eslintrc
   node_modules/.bin/eslint --reset -c ./eslintrc $(git --no-pager diff --name-only -M100% --diff-filter=AM --relative $(git merge-base FETCH_HEAD origin/HEAD) FETCH_HEAD | grep "\.js$")
 }
 

--- a/package.json
+++ b/package.json
@@ -32,10 +32,10 @@
     "assert-diff": "^1.0.1",
     "babel": "^5.5.4",
     "babel-core": "^5.5.4",
+    "babel-eslint": "^3.1.20",
     "babel-loader": "^5.0.0",
     "coveralls": "~2.10.0",
     "eslint": "^0.24.0",
-    "esprima-fb": "^15001.1.0-dev-harmony-fb",
     "eventemitter2": "^0.4.14",
     "gulp": "~3.8.10",
     "gulp-bump": "~0.1.13",
@@ -55,7 +55,7 @@
     "prepublish": "npm run clean && npm run compile",
     "test": "istanbul test _mocha",
     "coveralls": "cat ./coverage/lcov.info | coveralls",
-    "lint": "if ! [ -f eslintrc ]; then curl -o eslintrc 'https://raw.githubusercontent.com/ripple/javascript-style-guide/es6/eslintrc'; echo 'parser: esprima-fb' >> eslintrc; fi; eslint --reset -c eslintrc src/",
+    "lint": "if ! [ -f eslintrc ]; then curl -o eslintrc 'https://raw.githubusercontent.com/ripple/javascript-style-guide/es6/eslintrc'; echo 'parser: babel-eslint' >> eslintrc; fi; eslint --reset -c eslintrc src/",
     "perf": "./scripts/perf_test.sh"
   },
   "repository": {

